pkg/proxy/ipvs/proxier.go

```diff
@@ -1949,7 +1949,7 @@ func (proxier *Proxier) syncEndpoint(svcPortName proxy.ServicePortName, onlyNode
 			klog.Errorf("Failed to parse endpoint: %v, error: %v", ep, err)
 			continue
 		}
-		portNum, err := strconv.Atoi(port)
+		portNum, err := utilnet.ParsePort(port, false)
```

Comment:
0 was previously allowed here, right? I'll defer to the networking/ipvs folks to answer whether a 0 port should be treated as an error at this point now

Reply:
I don't think 0 is valid here.  @andrewsykim for cross check

Reply:
I _think_  0 port is accepted by IPVS, but 0 port is interpreted as "use the same port as the virtual server" which is still wrong in most cases so I think treating as an error would be acceptable at this point.
```
root@ipvs-worker:/# ipvsadm -L -t 10.96.0.10:53
Prot LocalAddress:Port Scheduler Flags
  -> RemoteAddress:Port           Forward Weight ActiveConn InActConn
TCP  10.96.0.10:53 rr
  -> 10.244.0.2:53                Masq    1      0          0         
  -> 10.244.0.3:53                Masq    1      0          0         
root@ipvs-worker:/# ipvsadm -a -t 10.96.0.10:53 -r 10.0.0.1:0    # ADD real server with 0 port
root@ipvs-worker:/# ipvsadm -L -t 10.96.0.10:53
Prot LocalAddress:Port Scheduler Flags
  -> RemoteAddress:Port           Forward Weight ActiveConn InActConn
TCP  10.96.0.10:53 rr
  -> 10.0.0.1:53                  Route   1      0          0     # Real server has port 53, matching virtual server         
  -> 10.244.0.2:53                Masq    1      0          0         
  -> 10.244.0.3:53                Masq    1      0          0 
```

I'm not sure we've actually passed in a 0 port here though. Trying to think of scenarios where this could happen.

Reply:
Either way I think not allowing zero port is correct here since we should never assume the service and endpoints ports are the same anyways.